### PR TITLE
feat(x/ecocredit)!: specify basket denom conventions

### PR DESCRIPTION
**Pull request by aaronc**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by ryanchristo — APPROVED** [review]
This format makes sense to me.

**Comment by aaronc** [comment]
I also renamed `credit_type_name` -> `credit_type_abbrev` based on our discussions

**Review by technicallyty — APPROVED** [review]

File: proto/regen/ecocredit/basket/v1/tx.proto [hunk 1]
```diff
@@ -28,12 +28,22 @@ message MsgCreate {
...
+  // `eco/<prefix><name>` and its display name will be `eco/<name>`.
   string name = 2;
 
-  // display_name will be used to create a bank Metadata display name for this
-  // basket token.
-  string display_name = 3;
+  // prefix is a single alphabetic character that denotes
+  // the denom prefix and indicates in some way what relationship is between
+  // the denom and display denom. Standard conventions are that m indicates
+  // milli (10^3), u indicates micro (10^6), n indicates nano (10^9).
```
Comment (clevinson) [inline proto/regen/ecocredit/basket/v1/tx.proto:40]:
```suggestion
  // milli (10^3), u indicates micro (10^6), n indicates nano (10^9).
  // Typical usage of prefix should correspond with the `exponent` field, 
  // which is used to convert decimal ecocredit amounts to integral
  // basket tokens.
```

**Review by clevinson — APPROVED** [review]
small comment upgrade. Otherwise lgtm

**Review by robert-zaremba — APPROVED** [review]

**Comment by clevinson** [comment]
@mergifyio backport release/v2.2.x

**Comment by S4mmyb** [comment]
This all looks good to me - thanks @aaronc!

**Pull request merged**